src/tmt_web/generators/json_generator.py

```diff
@@ -1,73 +1,150 @@
-import json
 from typing import Any
 
-import tmt.utils
-from tmt import Plan, Test
+from pydantic import BaseModel, ConfigDict, Field
+from tmt import Logger, Plan, Test
+from tmt.utils import GeneralError
 
 
-def _create_json_data(obj: Test | Plan, logger: tmt.Logger) -> dict[str, Any]:
+class FmfIdModel(BaseModel):
+    """Represents fmf-id data in JSON output."""
+
+    name: str
+    url: str | None = None
+    path: str | None = None
+    ref: str | None = None
+
+    @classmethod
+    def from_fmf_id(cls, fmf_id: Any) -> "FmfIdModel":
+        """Create FmfIdModel from a tmt FmfId object."""
+        return cls(
+            name=fmf_id.name,
+            url=fmf_id.url,
+            path=fmf_id.path.as_posix() if fmf_id.path is not None else None,
+            ref=fmf_id.ref,
+        )
+
+    def model_dump(self, **kwargs: Any) -> dict[str, Any]:
```

Comment:
I don't see a difference from the original `model_dump` generated by `pydantic`.